### PR TITLE
add iterator type to get_grouped_daily_aggs return

### PR DESCRIPTION
**Pull request by tobealive**
Running into a type error that doesn't seem like it should be an error, I'm assuming this return type should be iterable. Like other similar returns give a collection of data.

the comments below describe it as `List` so something is most definitely off.
`:return: List of grouped daily aggregates`

### CODE REVIEW

**Comment by tobealive** [comment]
hey @danielatpolygonio, thanks for covering this.
ofc. 
this is the error:

Screenshot_20221206_202626
in nvim with pyright.  Doubled checked with vscode, then it pops up as soon as turning `Python > Analysis: Type Checking Mode` to `basic`

```
Pyright: "GroupedDailyAgg" is not iterable
  "__iter__" method not defined [reportGeneralTypeIssues]
```

File: polygon/rest/aggs.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def get_grouped_daily_aggs(
         raw: bool = False,
         market_type: str = "stocks",
         include_otc: bool = False,
-    ) -> Union[GroupedDailyAgg, HTTPResponse]:
+    ) -> Union[Iterator[GroupedDailyAgg], HTTPResponse]:
```
Comment (jbonzo) [inline polygon/rest/aggs.py:63]:
```suggestion
    ) -> Union[List[GroupedDailyAgg], HTTPResponse]:
```

`List` is what we define for [`get_aggs`](https://github.com/polygon-io/client-python/blob/2bbb5c7f8a5d52641ec688fdf560ca3785dbb5a4/polygon/rest/aggs.py#L22) so we should be consistent with that.

Comment (tobealive) [reply polygon/rest/aggs.py:63]: 👍

**Comment by jbonzo** [comment]
@tobealive Thanks for the contribution! We appreciate it.

**Pull request merged**